File: problems/Custom Sort String/customSortString.cs
public class Solution {
    public string CustomSortString(string S, string T) {
        StringBuilder sb = new StringBuilder(T.Length);
        int[] tChars = new int[26];

        foreach (var tChar in T) {
            tChars[tChar - 'a']++;
        }

        foreach (var sc in S) {
            sb.Append(sc, tChars[sc - 'a']);
            tChars[sc - 'a'] = 0;
        }

        for (int i = 0; i < tChars.Length; i++) {
            sb.Append((char) (i + 'a'), tChars[i]);
        }

        return sb.ToString();
    }
}

File: problems/Custom Sort String/customSortString.java
class Solution {
    public String customSortString(String order, String str) {
        StringBuilder sb = new StringBuilder(str.length());
        int[] letterFrequencies = new int[26];

        for (char letter : str.toCharArray()) {
            ++letterFrequencies[letter - 'a'];
        }

        for (char letter : order.toCharArray()) {
            while (0 < letterFrequencies[letter - 'a']--) {
                sb.append(letter);
            }
        }

        for (int idx = 0; letterFrequencies.length > idx; ++idx) {
            while (0 < letterFrequencies[idx]--) {
                sb.append((char) ('a' + idx));
            }
        }

        return sb.toString();
    }
}
